NumLib/NumericsConfig.h

```diff
@@ -12,6 +12,8 @@
 
 #include <type_traits>
 
+#include "MathLib/LinAlg/SparsityPattern.h"
```

Comment:
Basically I would like to keep a module as general as possible. It's hard to do it for NumLib but MathLib is still possible with this way. My understanding is that these configuration are specifying how we use or specialize MathLib for our purpose, i.e. ogs6.